### DISCUSSION TITLE
How to avoid SonarQube and IntelliJ Plugin version mismatch

### DISCUSSION THREAD

**Discussion by Balaavchandran** [spotbugs/spotbugs#2766] [Q&A] [answered]
Have install SonarQube FindBugs plugin version 4.2.6 and in the IntelliJ IDE install SpotBugs version 1.2.5. ran the same Java project in IDE as well as SonarQube. When compared both the report,  the issue count between SonarQube and IntelliJ IDE comes around 20% (IntelliJ issue count higher than SonarQube). This count difference might be due to version mismatch between SonarQube and IntelliJ Plugins, what i have mentioned above. how to find equivalent version plugin for both Intellij and SonarQube to avoid this count difference. please help 

Thanks,
Balachandran A.V

**Comment by JuditKnoll** [comment, marked as answer]
It looks like the [SpotBugs IntelliJ Plugin](https://github.com/JetBrains/spotbugs-intellij-plugin) latest version, [1.2.5](https://github.com/JetBrains/spotbugs-intellij-plugin/releases/tag/1.2.5) is stuck on SpotBugs 4.4.2, while the [SonarQube FindBugs plugin](https://github.com/spotbugs/sonar-findbugs) is quite up to date, the [4.2.6](https://github.com/spotbugs/sonar-findbugs/releases/tag/4.2.6) version supports SpotBugs version 4.8.2.
If your goal is to use the same SpotBugs core versions, then you can downgrade the SonarQube plugin to match the IntelliJ plugin's version, or rather instead of the IntelliJ plugin, you can use SpotBugs via other integrations (ant, maven, gradle, eclipse), or you can also contribute to these projects, since all of them are opensource projects. However, the SpotBugs IntelliJ Plugin is developed by JetBrains, not by the SpotBugs Team, and I have no info whether it will be maintained.

**Reply by Balaavchandran** [reply]
Thanks a lot. so, will downgrade SonarQube FindBugs version to 4.0.4 means SpotBugs version 4.4.0, which might be close to SpotBugs IntelliJ plugin version 4..4.2. Will try and update this group. Much appreciate the real quick response and thanks again.
